language-support/hs/bindings/src/DA/Ledger/Services/CommandService.hs

```diff
@@ -41,14 +38,13 @@ submitAndWaitForTransactionId commands =
         let LL.CommandService{commandServiceSubmitAndWaitForTransactionId=rpc} = service
         let request = LL.SubmitAndWaitRequest (Just (lowerCommands commands)) noTrace
         rpc (ClientNormalRequest request timeout emptyMdm)
+            >>= unwrapWithInvalidArgument
             >>= \case
-            ClientNormalResponse response _m1 _m2 _status _details -> do
+            Right response -> do
```

Comment:
This pattern seems to be repeated quite often. Maybe we should factor it out into `(a -> IO b) -> Either StatusDetails a -> IO (Either String b)`? Or given that it seems like the only thing we ever do to the `StatusDetails` field is to call `show . unStatusDetails` on it, this could also be moved to `unwrapWithInvalidArgument` and the helper for this turns into `traverse`.

Reply:
Thanks. I moved the call to  `unStatusDetails` into `unwrapWithInvalidArgument`
and made some other trivial simplifications.
Next PR